### PR TITLE
feat(flux): implement rewrite rule for group-window-aggregate

### PR DESCRIPTION
**Pull request by lesam**
Part of #21069 .

Backports the group-window-aggregate rewrite rule to 1.x

- [x] [CHANGELOG.md](https://github.com/influxdata/influxdb/blob/master/CHANGELOG.md) updated with a link to the PR (not the Issue)
- [x] [Well-formatted commit messages](https://www.conventionalcommits.org/en/v1.0.0-beta.3/)
- [x] Rebased/mergeable
- [x] Tests pass

### CODE REVIEW

**Comment by lesam** [comment]
Depends on #21124 , which should be merged first.

**Comment by danxmoran** [comment]
Changed the PR base for now to make reviewing easier

**Review by danxmoran — COMMENTED** [review]
Can you add a new e2e test? It'll guard against the rule being removed from the top-level list in the future

File: flux/stdlib/influxdata/influxdb/rules_test.go [hunk 1]
```diff
@@ -1939,6 +1939,513 @@ func TestPushDownWindowAggregateRule(t *testing.T) {
...
+			},
+		},
+	})
+
+	// 2.x has a feature-off case, but we don't support feature flagging so we ignore it.
+
+	for _, tc := range tests {
+		tc := tc
+		t.Run(tc.Name, func(t *testing.T) {
+			//t.Parallel()
```
Comment (danxmoran) [inline flux/stdlib/influxdata/influxdb/rules_test.go:2443]: Does something break if we run in parallel?

File: flux/stdlib/influxdata/influxdb/rules.go [hunk 2]
```diff
@@ -296,20 +297,6 @@ func (rule PushDownReadTagKeysRule) Rewrite(ctx context.Context, pn plan.Node) (
 	}), true, nil
 }
 
-func hasFieldExpr(expr semantic.Expression) bool {
```
Comment (lesam) [inline flux/stdlib/influxdata/influxdb/rules.go:299]: This was unused

File: flux/stdlib/influxdata/influxdb/rules_test.go (hunk 1 above)
Comment (lesam) [reply flux/stdlib/influxdata/influxdb/rules_test.go]: Nope, this was a hack for testing - oops.

**Comment by lesam** [comment]
@jsternberg says this is not yet ready for primetime, so let's not backport it.

**Pull request closed without merging**